internal/gofumpt.go

```diff
@@ -516,6 +528,88 @@ func (f *fumpter) joinStdImports(d *ast.GenDecl) {
 	}
 }
 
+// mergeAdjacentFields mutates fieldList to merge adjacent fields if possible
+// and returns fieldList.
+func (f *fumpter) mergeAdjacentFields(fieldList *ast.FieldList) *ast.FieldList {
+	// If there are less than two fields then there is nothing to merge.
+	if len(fieldList.List) < 2 {
+		return fieldList
+	}
+
+	// Otherwise, iterate over adjacent pairs of fields, merging if possible,
+	// and building a new list. Elements of fieldList.List may be mutated (if
+	// merged with following fields), discarded (if merged with a preceeding
+	// field), or left unchanged.
+	lastField := fieldList.List[0]
+	newList := []*ast.Field{lastField}
+	mutated := false
+	for i := 1; i < len(fieldList.List); i++ {
+		field := fieldList.List[i]
+		if f.shouldMergeAdjacentFields(field, lastField) {
+			lastField.Names = append(lastField.Names, field.Names...)
+			mutated = true
+		} else {
+			newList = append(newList, field)
+			lastField = field
+		}
+	}
+
+	// Mutate fieldList if the new list is different.
+	if mutated {
+		fieldList.List = newList
+	}
+
+	return fieldList
+}
+
+func (f *fumpter) shouldMergeAdjacentFields(f1, f2 *ast.Field) bool {
+	// Do not merge if either field has doc.
+	if f1.Doc != nil || f2.Doc != nil {
+		return false
+	}
+
+	// Do not merge if either field has no names.
+	if len(f1.Names) == 0 || len(f2.Names) == 0 {
+		return false
+	}
+
+	// Do not merge if the fields span more than one line. Check both fields in
+	// case f2 preceeds f1.
+	if line := f.Line(f1.Pos()); f.Line(f1.End()) != line || f.Line(f2.Pos()) != line || f.Line(f2.End()) != line {
```

Comment:
Go doesn't support fields split in multiple lines, like

```
func f(
    name
    Type,
)
```

The only case we want to worry about is

```
func f(
    name1 Type,
    name2 Type,
)
```

So I'm pretty sure `if f.Line(f1.Pos()) != f.Line(f2.Pos())` should be enough.

Reply:
Done.